src/main/resources/default/taglib/w/deleteLink.html.pasta

```diff
@@ -1,12 +1,16 @@
-<i:arg type="String" name="url" />
+<i:arg type="String" name="url"/>
 <i:arg type="boolean" name="adminOnly" default="false"/>
 <i:arg type="Page" name="page" default=""/>
 
-<i:pragma name="inline" value="true" />
-<i:pragma name="description" value="Renders a delete/dangerous link within a Wondergem template" />
+<i:pragma name="inline" value="true"/>
+<i:pragma name="description" value="Renders a delete/dangerous link within a Wondergem template"/>
 
-<i class="fa fa-trash"></i>
-<a class="link link-danger @if (adminOnly) {admin-link}"
-   href="@if(page != null) {@url?@page.createQueryString()} else {@url}">
-    @i18n("NLS.delete")
-</a>
+<i:local name="deleteFormId" value="@apply('deleteForm-%s', call.generateLocalId())"/>
+
+<form id="@deleteFormId" method="post" action="@if(page != null) {@url?@page.createQueryString()} else {@url}">
+    <i class="fa fa-trash"></i>
+    <a class="link link-danger @if (adminOnly) {admin-link}" data-deleteid="@deleteFormId" href="#">
+        @i18n("NLS.delete")
+    </a>
+    <input name="CSRFToken" value="@call.getCSRFToken()" type="hidden"/>
```

Comment:
add CSRFHelper to DefaultGlobalContextExtender as "csrf" and call csrf.getToken(call) here